lib/hci-socket/gatt.js

```diff
@@ -743,4 +741,81 @@ Gatt.prototype.writeHandle = function(handle, data, withoutResponse) {
   }
 };
 
+Gatt.prototype.handleRequest = function(request) {
+  debug('handing request: ' + request.toString('hex'));
+
+  var requestType = request[0];
+  var response = null;
+
+  switch(requestType) {
+    case ATT_OP_FIND_BY_TYPE_REQ:
+      response = this.handleFindByTypeRequest(request);
+      break;
+
+    default:
+    case ATT_OP_READ_MULTI_REQ:
+    case ATT_OP_SIGNED_WRITE_CMD:
+      debug(this._address + ': replying with REQ_NOT_SUPP to 0x' + requestType.toString(16));
+      response = this.errorResponse(requestType, 0x0000, ATT_ECODE_REQ_NOT_SUPP);
+      break;
+  }
+
+  if (response) {
+    debug('response: ' + response.toString('hex'));
+
+    this.writeAtt(response);
+  }
+};
+
+Gatt.prototype.handleFindByTypeRequest = function(request) {
+  var response = null;
+
+  var startHandle = request.readUInt16LE(1);
+  var endHandle = request.readUInt16LE(3);
+  var uuid = request.slice(5, 7).toString('hex').match(/.{1,2}/g).reverse().join('');
+  var value = request.slice(7).toString('hex').match(/.{1,2}/g).reverse().join('');
+
+  var handles = [];
+  var handle;
+
+  for (var i = startHandle; i <= endHandle; i++) {
+    handle = this._handles[i];
```

Comment:
Where does this._handles get populated with the actual handles?

Reply:
@CTunstall It doesn't.

Reply:
@anurag-k4 Those two handles[] arrays are not related.  The one in bindings.js refers to peripheral handles, but the one in gatt.js was cribbed from bleno (sister project to noble), and refers to services that are registered with a peripheral.  Since we are not implementing a bluetooth peripheral, there will be no services registered.